x-pack/plugins/siem/public/common/components/add_filter_to_global_search_bar/index.test.tsx

```diff
@@ -32,11 +33,10 @@ jest.mock('../../lib/kibana', () => ({
 }));
 
 describe('AddFilterToGlobalSearchBar Component', () => {
-  const state: State = mockGlobalState;
-  let store = createStore(state, SUB_PLUGINS_REDUCER, apolloClientObservable);
+  let store: SecurityAppStore;
 
   beforeEach(() => {
-    store = createStore(state, SUB_PLUGINS_REDUCER, apolloClientObservable);
+    store = createStore(mockGlobalState, SUB_PLUGINS_REDUCER, apolloClientObservable);
```

Comment:
`createStore` takes a 'preloadedState' param, which is passed through to redux's `createStore` as the `preloadedState` param. 

The `prelodadeState` param to redux's `createStore` is of type `PreloadedState<S>` where `S` is the redux state's type. `PreloadedState` makes any part of the state defined with `combineReducers` optional. This is because you might have preloaded state for one part of your state but not another.

`mockGlobalState` satisfies `PreloadedState<State>` so it can be used to create the store, but it doesn't satisfy `State`, because it doesn't have preloaded state for the new endpoint subplugins.

These tests typically imported `mockGlobalState` and set it to a binding like 
```
const state: State = mockGlobalState;
```

By removing these bindings which have a type of `State`, and passing `mockGlobalState` directly to `createStore`, the types are all good. 

Note that a few tests do `mockGlobalState` in another way. They pull off specific keys and use that data. For this reason, `mockGlobalState` is typed as `PreloadedState<State> & otherStuff` where other stuff is a type that provides the guarantees that certain tests expect.

The endpoint subplugins will aim not to rely on global state for their tests.